backend/src/main/java/com/example/dontwastetomuch/game/GameController.java

```diff
@@ -22,40 +24,56 @@ public class GameController {
 
     @PostMapping()
     @ResponseStatus(HttpStatus.CREATED)
-    public void addGame(@RequestBody Game game, Principal principal){
-        MyUser user = myUserRepo.findById(principal.getName()).orElseThrow();
-        gameService.addGame(game, user);
+    public ResponseEntity<Void> addGame(@RequestBody Game game, Principal principal){
+        try {
+            MyUser user = myUserRepo.findById(principal.getName()).orElseThrow();
+            gameService.addGame(game, user);
+            return ResponseEntity.status(HttpStatus.CREATED).build();
+        } catch (IllegalArgumentException | IllegalStateException e){
+            return new ResponseEntity(e.getMessage(), HttpStatus.METHOD_NOT_ALLOWED);
+        }
```

Comment:
n1 using this constructor. This way the body type of ResponseEntity in the method return can be the real one for success and not object